apps/studio/src/server/modules/resource/__tests__/resource.router.test.ts

```diff
@@ -0,0 +1,1722 @@
+      })
+
+      // Act
+      const result = caller.move({
+        movedResourceId: pageToMove.id,
+        destinationResourceId: anotherPage.id,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "BAD_REQUEST" }),
+      )
+    })
+
+    it("should return 403 if source and destination resources belong to different sites", async () => {
+      // Arrange
+      const { page: originPage, site: originSite } = await setupPageResource({
+        resourceType: "Page",
+      })
+      const { folder: destinationFolder, site: destinationSite } =
+        await setupFolder()
+      expect(originSite.id).not.toEqual(destinationSite.id)
+
+      // Act
+      const result = caller.move({
+        movedResourceId: originPage.id,
+        destinationResourceId: destinationFolder.id,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "FORBIDDEN" }),
+      )
+    })
+
+    it("should move nested resource to destination folder", async () => {
+      // Arrange
+      const { folder: originFolder, site } = await setupFolder({
+        permalink: "origin-folder",
+      })
+      const { page: pageToMove } = await setupPageResource({
+        resourceType: "Page",
+        siteId: site.id,
+        parentId: originFolder.id,
+      })
+      const { folder: destinationFolder } = await setupFolder({
+        siteId: site.id,
+        permalink: "destination-folder",
+      })
+
+      // Act
+      const result = await caller.move({
+        movedResourceId: pageToMove.id,
+        destinationResourceId: destinationFolder.id,
+      })
+
+      // Assert
+      const expected = {
+        ...pick(pageToMove, ["id", "type", "permalink", "title"]),
+        parentId: destinationFolder.id,
+      }
+      const actual = await db
+        .selectFrom("Resource")
+        .where("id", "=", pageToMove.id)
+        .select(["parentId"])
+        .executeTakeFirstOrThrow()
+      expect(actual.parentId).toEqual(destinationFolder.id)
+      expect(result).toMatchObject(expected)
+    })
+
+    it("should move root-level resource to destination folder", async () => {
+      // Arrange
+      const { page: pageToMove, site } = await setupPageResource({
+        resourceType: "Page",
+        parentId: null,
+      })
+      const { folder: destinationFolder } = await setupFolder({
+        siteId: site.id,
+        permalink: "destination-folder",
+      })
+
+      // Act
+      const result = await caller.move({
+        movedResourceId: pageToMove.id,
+        destinationResourceId: destinationFolder.id,
+      })
+
+      // Assert
+      const expected = {
+        ...pick(pageToMove, ["id", "type", "permalink", "title"]),
+        parentId: destinationFolder.id,
+      }
+      expect(result).toMatchObject(expected)
+    })
+
+    it.skip("should throw 403 if user does not have write access to destination resource", async () => {})
```

Comment:
we might need 1 more (skipped) test case: admin can move folders/pages from anywhere to root